### PR TITLE
Add queries to CLI and Account interface

### PR DESCRIPTION
**Pull request by ericnordelo**
Drafted because it lacks test and documentation but is functional already. I will add the rest tomorrow.

It can be used from the NRE, but the methods are returning a string (instead of the estimated gas fee, for example) because the current design of the CLI prints from this output instead of logging properly (output needs to be parsed from NRE).

A refactor to use loggers properly would allow us to return the proper values for NRE use (like tx_hash in Account.send instead of the full string from starknet.cli).

### CODE REVIEW

**Review by andrew-fleming — DISMISSED** [review]
Great work, @ericnordelo! I just left a tiny suggestion :) let's also keep in mind that we'll have to add a `version` arg to Contracts's signer methods before releasing Contracts

File: src/nile/cli.py [hunk 1]
```diff
@@ -117,18 +117,26 @@ def setup(signer, network):
 @click.argument("method", nargs=1)
```
Comment (martriay) [inline src/nile/cli.py:117]: why the change in the permissions? (`100755` → `100644`)

File: src/nile/core/account.py [hunk 2]
```diff
@@ -79,29 +79,58 @@ def deploy(self):
...
-            max_fee = 0
-        else:
-            max_fee = int(max_fee)
+    def estimate_fee(self, address_or_alias, method, calldata, max_fee, nonce=None):
+        """Estimate fee for a tx going through an Account contract."""
+        return self.execute_query(
+            "estimate_fee", address_or_alias, method, calldata, max_fee, nonce
+        )
+
+    def execute_query(
```
Comment (martriay) [inline src/nile/core/account.py:118]: what do you think about renaming this `execute` or something similar, and to also accept a `transaction` type that actually executes it, to avoid code duplication with `send`

File: src/nile/core/call_or_invoke.py [hunk 3]
```diff
@@ -10,9 +10,27 @@
...
+    Call or invoke functions of StarkNet smart contracts.
+
+    @param contract: can be an address, an alias, or an Account instance.
+    @param type: can be either call or invoke.
+    @param method: the targeted function.
+    @param params: the targeted function arguments.
+    @param network: goerli, mainnet, or predefined networks file.
+    @param signature: optional signature for invoke transactions.
+    @param max_fee: optional max fee for invoke transactions.
+    @param query_flag: either simulate or estimate_fee
```
Comment (martriay) [inline src/nile/core/call_or_invoke.py:32]: 👍

File: tests/commands/test_account.py [hunk 4]
```diff
@@ -111,3 +116,69 @@ def test_send_sign_transaction_and_execute():
...
+    # Mock sign_transaction
+    account.signer.sign_transaction = MagicMock(return_value=return_signature)
+
+    account.execute_query(query_type, account.address, "method", [1, 2, 3], max_fee=0)
+
+    account.signer.sign_transaction.assert_called_once_with(
+        calls=[send_args],
+        nonce=0,
+        sender=account.address,
+        max_fee=0,
```
Comment (martriay) [inline tests/commands/test_account.py:170]: do you think we should pass a `max_fee` different than `0` to make sure it's going through, considering `0` is the default?

**Review by martriay — APPROVED** [review]
Looks good! Left some comments

File: src/nile/cli.py (hunk 1 above)
Comment (ericnordelo) [reply src/nile/cli.py:117]: Not sure at what moment the permission was updated, to be honest. Fixed again to 755.

**Review by martriay — APPROVED** [review]
Please resolve conflicts and good to go!

**Review by martriay — DISMISSED** [review]

**Review by andrew-fleming — APPROVED** [review]
Great work, @ericnordelo! Looks good to me :)

**Review by martriay — APPROVED** [review]
